README.md

```diff
@@ -538,6 +538,71 @@ api.addSubscription('addedFilm', new Field({
 ```
 
 To learn more about top level operations, check out the docs [here](https://docs.aws.amazon.com/appsync/latest/devguide/real-time-data.html).
+
+## Merge Source API to Merged API Using A Custom Resource
+
+The SourceApiAssociationMergeOperation construct provides the ability to merge a source api to a Merged Api
+and invoke a merge within a Cloudformation custom resource. If the merge operation fails with a conflict, the 
+Cloudformation update will fail and rollback the changes to the source API in the stack.
+
+```ts
+import * as cdk from 'aws-cdk-lib';
+
+const sourceApi1ToMerge = new appsync.GraphqlApi(this, 'FirstSourceAPI', {
+  name: 'FirstSourceAPI',
+  definition: appsync.Definition.fromFile(path.join(__dirname, 'appsync.merged-api-1.graphql')),
+});
+
+const sourceApi2ToMerge = new appsync.GraphqlApi(this, 'SecondSourceAPI', {
+  name: 'SecondSourceAPI',
+  definition: appsync.Definition.fromFile(path.join(__dirname, 'appsync.merged-api-2.graphql')),
+});
+
+const remoteMergedApi = appsync.GraphqlApi.fromGraphqlApiAttributes(this, 'ImportedMergedApi', {
+  graphqlApiId: 'MyApiId',
+  graphqlApiArn: 'MyApiArn',
+});
+
+const remoteExecutionRole = iam.Role.fromRoleArn(this, 'ExecutionRole', 'arn:aws:iam::ACCOUNT:role/MyExistingRole');
+const association1 = new appsync.SourceApiAssociation(this, 'SourceApiAssociation2', {
+   sourceApi: sourceApi1ToMerge,
+   mergedApi: remoteMergedApi,
+   mergeType: appsync.MergeType.MANUAL_MERGE,
+   mergedApiExecutionRole: remoteExecutionRole,
+});
+
+const association2 = new appsync.SourceApiAssociation(this, 'SourceApiAssociation2', {
+   sourceApi: sourceApi2ToMerge,
+   mergedApi: remoteMergedApi,
+   mergeType: appsync.MergeType.MANUAL_MERGE,
+   mergedApiExecutionRole: remoteExecutionRole,
+});
+
+// The SourceApiAssociationMergeOperationProvider construct creates the Lambda handlers for submitting the merge operation 
+// and ensuring that it succeeded or failed.
+const provider = new appsync.SourceApiAssociationMergeOperationProvider(this, 'SchemaMergeProvider', {
```

Comment:
I'm wondering why the user has to instantiate the provider here? Can't we just do it for them if no provider is passed to `SourceApiAssociationMergeOperation`?

Reply:
Yea, i can make that change. The idea is they can pass 1 provider to a bunch of these custom resources so they dont have too many Lambdas that end up getting created.

Reply:
Usually we use a `SingletonLambda` for custom resource providers to avoid deploying multiples into user accounts anyway. We have some utilities to help with this `CustomResourceProvider.getOrCreateProvider`, handles this but it doesn't handle building/bundling like NodeJS function. I would add some mechanism for only creating the function once within a stack. You can just cache the instantiation of the function somewhere and check it's existence.

Reply:
Yea, that also does not work with Provider framework. Can you clarify what you mean by caching the instantiation of the function?